pynumaflow/sink/server.py

```diff
@@ -57,11 +56,20 @@ def SinkFn(
         The pascal case function name comes from the generated udsink_pb2_grpc.py file.
         """
 
-        msgs = self.__sink_handler(request.elements)
+        try:
+            rspns = self.__sink_handler(request.elements)
+        except Exception as err:
+            err_msg = "UDSinkError: %r" % err
+            _LOGGER.exception(err_msg)
+            rspns = Responses()
+            for _datum in request.elements:
+                rspns.append(Response.as_failure(_datum.id, err_msg))
 
         responses = []
-        for msg in msgs.items():
-            responses.append(udsink_pb2.Response(id=msg.id, success=msg.success, err_msg=msg.err))
+        for rspn in rspns.items():
+            responses.append(
+                udsink_pb2.Response(id=rspn.id, success=rspn.success, err_msg=rspn.err)
```

Comment:
@vigith this part is good, actually. We are creating a new list `responses` from `rspns` regardless of the exception or not.